NEventSocket/Sockets/Parser.cs

```diff
@@ -98,8 +95,10 @@ public Parser Append(char next)
             }
             else
             {
+                Debug.Assert(bodyBytes != null);
+                bodyBytes.WriteByte(next);
                 // if we've read the Content-Length amount of bytes then we're done
-                Completed = buffer.Length == contentLength.GetValueOrDefault() || contentLength == 0;
+                Completed = bodyBytes.Length == contentLength.GetValueOrDefault() || contentLength == 0;
```

Comment:
If `contentLength` would be `null` or `0` this code would never be reached because `HasBody` would be `false`. Therefore it's less confusing to use `bodyBytes.Length == contentLength` here and perhaps state the fact with a `Debug.Assert(contentLength > 0);` (this isn't actually a part of this PR, but it makes comprehending this function a bit easier).